Client/Frontend/Home/BookmarksEditPanel.swift

```diff
@@ -0,0 +1,335 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import UIKit
+import Storage
+import Shared
+import XCGLogger
+
+private let log = Logger.browserLogger
+
+enum DetailPanel: Int {
+    case Title = 0
+    case URL
+    case Folder
+    case _numberOfPanels
+}
```

Comment:
Using the last enum as the 'count' for the number of enum properties isn't very swift-y. Instead, I would add a static property on it containing the number of rows explicitly. 

Also I don't think the name `DetailPanel` is accurate for the way we're using this enum. Maybe something like `BookmarkEditField`?